synapse/handlers/auth.py

```diff
@@ -307,14 +309,26 @@ def _check_recaptcha(self, authdict, clientip):
                 defer.returnValue(True)
         raise LoginError(401, "", errcode=Codes.UNAUTHORIZED)
 
-    @defer.inlineCallbacks
     def _check_email_identity(self, authdict, _):
+        return self._check_threepid('email', authdict)
+
+    def _check_msisdn(self, authdict, _):
+        return self._check_threepid('msisdn', authdict)
+
+    @defer.inlineCallbacks
+    def _check_dummy_auth(self, authdict, _):
+        yield run_on_reactor()
+        defer.returnValue(True)
```

Comment:
you could do *that* with `return success(True)`.

A better answer might be: because it did before we got here, and changing it is scary.